service/api_task.go

```diff
@@ -223,7 +223,13 @@ func (as *APIServer) EndTask(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// mark task as finished
-	deactivatePrevious := utility.FromBoolPtr(projectRef.GetProjectParserMergedProjectRef(t.Version).DeactivatePrevious)
+	mergedProjectRef, err := projectRef.MergeWithParserProject(t.Version)
+	if err != nil || mergedProjectRef == nil {
```

Comment:
We can't handle these errors together. If err == nil and mergedProjectRef == nil, then 228 will be nil because there is no error to wrap, and we'll be silently erroring.

Reply:
Good catch, thanks